### ISSUE TITLE
Application Configurations not working

### ISSUE THREAD

**Issue by Martos** [h-mdm/hmdm-server#49]
Hello, i trying to define an app restriction of type string, but the bundle return empty. I used this guide to create a listener https://developer.android.com/work/managed-configurations

**Comment by Martos** [comment]
![mdm](https://user-images.githubusercontent.com/4572919/172845933-198b6047-7f54-4738-927a-ff1cc9252863.PNG)

**Comment by h-mdm** [comment]
Headwind MDM cannot use managed configurations API because it's not a certified Google EMM provider.

Here's how Application Settings work: https://qa.h-mdm.com/1121/

**Comment by h-mdm** [comment]
Partnership with Samsung Knox is not related to Android Enterprise or Google EMM Provider, so the Android Enterprise SDK is still not available, sorry about that.

**Comment by nktnet1** [comment]
Thanks @h-mdm.

Something I don't understand is that apps such as

1. OwnDroid: https://github.com/BinTianqi/OwnDroid

    - code: https://github.com/BinTianqi/OwnDroid/commit/4bcd2d8150c7371c228b3125437c96dfb4e7959f

2. Test DPC: https://github.com/googlesamples/android-testdpc
    
    - code: https://github.com/googlesamples/android-testdpc/blob/305c86d36e9b7c85e41e9e01fbf2d648fa4970fb/src/main/java/com/afwsamples/testdpc/ShellCommand.java#L1204-L1229

Are all able to set **managed configurations** (app restrictions) simply by being the Device Owner.

Given that the headwind android app can be set as the device owner, is there any other limitations I'm unaware of?

**Comment by h-mdm** [comment]
You are right, this feature is definitely worth being implemented - we were misleaded by the method name "setApplicationRestrictions" whereas it is used to configure (not restrict) third party apps, so a device owner app indeed can manage third party app configurations! 

However, we have to think how to integrate this option in the application settings so the existing application settings scheme won't be broken.

If you already implemented this feature and want to push your update as a pull request, that would be helpful for our dev team, so we'd be happy to get it. Don't worry if there will be no direct acceptance of the pull request, your contribution will be acknowledged and processed anyway.

An alternative would be to wait until this feature is implemented.
